Fix jenkins integration

andrunko opened the pull request:
This fixes Endless Jenkins integration that got broken after 064fdeaa85e00bf72832d1991a571b4e009137ae.

~~It should still allow running the container as it is now but with some small caveats (e.g. if passing only a manifest as param to "docker run" it has to end with ".json", ".yml" or ".yaml").~~

andrunko committed:
Re-add support for running container with arbitrary internal uid:gid

We pass the user and group ids when building the image used by
CI/Jenkins as a workaround to make sure the user running the container
exists inside of it (see Jenkinsfile/Dockerfile). This is needed because
jenkins runs the container with the same user as the host
(with '-u <uid>:<gid>') but 'git' fails with 'No user exists for uid ...'
if the user doesn't exist inside the container.

See also https://issues.jenkins-ci.org/browse/JENKINS-47026.

gasinvein commented on Dockerfile:
```diff
@@ -24,10 +24,20 @@ RUN apt-get update && \
 RUN python3 -m compileall /app/src
 
-ENTRYPOINT [ "/app/flatpak-external-data-checker" ]
+ENTRYPOINT [ "/docker-entrypoint.sh" ]
```
Do you control docker cli args in Jenkins? If so, maybe it's better to pass `--entrypoint whatever` to `docker` instead of guessing what we should run in a shell script?

andrunko replied:
Not sure tbh, the error comes from (which runs `cat` to test the container):
```
$ docker run -t -d -u 1004:1004 --privileged -w /home/jenkins/workspace/check-flatpak-external-apps-pipeline -v /home/jenkins/workspace/check-flatpak-external-apps-pipeline:/home/jenkins/workspace/check-flatpak-external-apps-pipeline:rw,z -v /home/jenkins/workspace/check-flatpak-external-apps-pipeline@tmp:/home/jenkins/workspace/check-flatpak-external-apps-pipeline@tmp:rw,z -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** flatpak-external-data-checker:latest cat
$ docker top 54624e2f05c7096d62a00b012b29e5ff3734393f613e885990d9c1c299799e9f -eo pid,comm
ERROR: The container started but didn't run the expected command. Please double check your ENTRYPOINT does execute the command passed as docker run argument, as required by official docker images (see https://github.com/docker-library/official-images#consistency for entrypoint consistency requirements).
```

I could try passing `--entrypoint` to `image.inside` on Jenkinsfile to see if that would work, but wonder if it may still be useful (and best practice) to have an entrypoint allowing running arbitrary commands (bash for example for debugging shell)?

gasinvein replied:
>it may still be useful (and best practice) to have an entrypoint allowing running arbitrary commands

Since the first argument is likely to be a file with arbitrary name, such wrapper feels a bit fragile. But I'm not strictly against it, it may be useful indeed.

andrunko replied:
Pushed a new branch `fix-jenkins-integration-new` overriding the entrypoint via Jenkinsfile and will verify if it works and update here accordingly.

andrunko replied:
Ok, it works, updated changes.

andrunko committed:
Jenkinsfile: Override default entrypoint

Jenkins will try running `cat` to test the container after building
it but the default entrypoint does not allow running arbitrary commands.
Let's not use an entrypoint when running on Jenkins.

gasinvein reviewed (approved):
Seems fine.

gasinvein commented on Dockerfile:
```diff
@@ -24,6 +24,14 @@ RUN apt-get update && \
 
+# Creating the user is required because jenkins runs he container
+# with the same user as the host (with '-u <uid>:<gid>')
+# but without the user existing 'git' fails with 'No user exists for uid ...'.
```
I wonder why `git` behaves like this - why does it even care about local system user presence?

andrunko replied:
Good question, no idea but it does...

andrunko replied:
Looking at the description on https://issues.jenkins.io/browse/JENKINS-47026, it seems not only git fails, ssh also (which is probably why git fails).
